packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -9,7 +13,15 @@ const createStubFileWatcher = (): ts.FileWatcher => ({
 
 export type TypeScriptProjectService = ts.server.ProjectService;
 
-export function createProjectService(): TypeScriptProjectService {
+export interface ProjectServiceSettings {
+  allowDefaultProjectForFiles: ReadonlySet<string>;
```

Comment:
This name is not particularly good but it's under an experimental option so I'd expect we'll rethink all names for v7.